### PR TITLE
Clean up temporal client and change the acceptance test harness to ca…

### PR DESCRIPTION
**Pull request by benmoriceau**
…ll the right method

## What
Extract the port forward of the airbyte temporal instance to a docker-compose file outside of the main docker-compose. 

This allow avoiding exposing it in all the docker instances and potentially to the open internet.

### CODE REVIEW

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Comment by jdpgrailsdev** [comment]
@benmoriceau  Looks like an acceptance test failed, but unsure if it is related to the changes in the PR:

```
BasicAcceptanceTests > testResetConnectionRepairsWorkflowWhenWorkflowUnreachable() FAILED
    java.lang.RuntimeException: Could not create Temporal client within max timeout!
        at io.airbyte.commons.temporal.TemporalUtils.getTemporalClientWhenConnected(TemporalUtils.java:225)
        at io.airbyte.commons.temporal.TemporalUtils.createTemporalService(TemporalUtils.java:95)
        at io.airbyte.test.utils.AirbyteAcceptanceTestHarness.getWorkflowClient(AirbyteAcceptanceTestHarness.java:318)
        at io.airbyte.test.utils.AirbyteAcceptanceTestHarness.terminateTemporalWorkflow(AirbyteAcceptanceTestHarness.java:333)
        at io.airbyte.test.acceptance.BasicAcceptanceTests.testResetConnectionRepairsWorkflowWhenWorkflowUnreachable(BasicAcceptanceTests.java:808)
```

**Comment by benmoriceau** [comment]
@colesnodgrass @jdpgrailsdev Yeah it shouldn't fail. My assumption is that the env was wrongly set. https://github.com/airbytehq/airbyte/pull/17864/commits/bc218721096078ea002800aa0097eccc5b3e8f44 is suppose to fix it. Trying locally takes too long and I am working on another issue, will wait for the build.

File: docker-compose.yaml [hunk 1]
```diff
@@ -165,8 +165,6 @@ services:
     logging: *default-logging
     container_name: airbyte-temporal
     restart: unless-stopped
-    ports:
-      - 7233:7233
```
Comment (evantahler) [inline docker-compose.yaml:169]: I'd suggest double checking that this also continues to work with the temporal UI that `docker-compose.debug.yaml` uses

**Review by evantahler — APPROVED** [review]

File: docker-compose.yaml (hunk 1 above)
Comment (benmoriceau) [reply docker-compose.yaml:169]: I will check that and add it to the debug before merging if needed

Comment (benmoriceau) [reply docker-compose.yaml:169]: @evantahler Tested locally, I didn't see any issues.

**Pull request merged**